hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -2070,4 +2071,27 @@ private String reconfOzoneAdmins(String newVal) {
         newVal, admins);
     return String.valueOf(newVal);
   }
+
+  /**
+   * This will remove the given SCM node from HA Ring by removing it from
+   * Ratis Ring and deleting the related certificates from certificate store.
+   *
+   * @return true if remove was successful, else false.
+   */
+  public boolean removePeerFromHARing(RemoveSCMRequest request)
+      throws IOException {
+    // We cannot remove a node if it's currently leader.
+    if (scmContext.isLeader() && request.getScmId().equals(getScmId())) {
```

Comment:
Q. If we need to perform maintenance on the _current leader_ SCM node, how do we decommission it?  

> // We cannot remove a node if it's currently leader.

Reply:
First, we have to stop the SCM service that we want to decommission. This will be part of SCM decommissioning documentation.